### DISCUSSION TITLE
Remix: Vyper plugin (remote compiler) is not working --> work around provided

### DISCUSSION THREAD

**Discussion by mat774** [vyperlang/vyper#2253] [Q&A]
Hello fellow friends of Vyper,

Not long ago, I was using remix to compile and test my smart contracs written in Vyper. I used the Vyper plugin (compiler), but recently it's not compiling my contracts anymore. 

Even a very simple contract is not compiling (using the remote compiler). 

![grafik](https://user-images.githubusercontent.com/38216165/101993266-75543000-3cb9-11eb-9af9-0729f995580e.png)
Simple test contract

![grafik](https://user-images.githubusercontent.com/38216165/101993312-cbc16e80-3cb9-11eb-9e29-0c506c49a3a9.png)

On the other hand, locally I'm able to compile every contract (so there is no error in the code).

![grafik](https://user-images.githubusercontent.com/38216165/101993330-fca1a380-3cb9-11eb-900c-45c2297808a4.png)

Does anybody know, why the remote compiler for vyper is not working in Remix?

However, for anybody who still wants to use Remix at the moment, you can easily connect to your local compiler.
Use the command vyper-serve to run the compiler and open port 8000 on you local mashine for incoming code.
![grafik](https://user-images.githubusercontent.com/38216165/101993415-c44e9500-3cba-11eb-8738-4a1b10f11ec2.png)

And then tell Remix to use your local compiler.
![grafik](https://user-images.githubusercontent.com/38216165/101993427-eea05280-3cba-11eb-9a3b-5e83aabdadf4.png)

I was searching the net for a solution and didn't found one, so I thought maybe another Vyper beginner such as me is looking for a workaround.

Greetings,
Matt

**Comment by kaleb-keny** [comment]
Honestly I used etherscan to get the byte code and abi it worked pretty well... Then pushed the byte code on myetherwallet

**Reply by arjunaskykok** [reply]
The link to etherscan's Vyper compiler: https://etherscan.io/vyper

However, I think we need to fix this issue because Remix interface is so nice. Someone needs to donate a server for this online Vyper compiler to be used with Remix.

**Comment by jacqueswww** [comment]
@mat774  Remix remote compiling has been up for a week or so again, have at it! :-)

**Reply by mat774** [reply]
Thanks for the info!
But somehow the deployment of contracts doesn't work anymore with compiled Vyper code.

Steps:
1) Create a .vy Vyper contract in Remix
2) Compile
3) Under the tab "Run & deploy contract" Remix doesn't recognise Vyper contracts anymore. Only Solidity.

Shall I create a own topic for this issue?
